File: TraceAndNormalMatrix/Find.py
import numpy as np;
import math;

def TraceOfMatrix(arr,irow,icol) :
    i = 0;
    j = 0;
    isum = 0;

    for i in range(irow) :
        for j in range(icol) :
            if(i == j) :
                isum = isum + arr[i][j];

    return isum;

def NormalOfMatrix(arr,irow,icol) :
    i = 0;
    j = 0;
    isum = 0;
    sqrt = 0;

    for i in range(irow) :
        for j in range(icol) :
            isum = isum + arr[i][j];

    sqrt = math.sqrt(isum);

    return sqrt;

def main() :
    i = 0;
    j = 0;
    irow = 0;
    icol = 0;
    arr = [[]];
    iret = 0;
    
    print("Jay Ganesh....");
    print("Enter number of rows : ");
    irow = int(input());

    print("Enter number of cols : ");
    icol = int(input());

    arr = np.zeros((irow,icol));
    print("Enter elements in matrix");
    for i in range(irow) :
        print("----Enter elements of %d row----"%(i + 1));
        for j in range(icol) :
            print("Enter element no %d : "%(j + 1),end="");
            arr[i][j] = int(input());

    print("Elements of matrix are : ");
    for i in range(irow) :
        for j in range(icol) :
            print("%d"%arr[i][j],end = "\t");
        print();

    iret = TraceOfMatrix(arr,irow,icol);
    print("Trace of matrix is : %d"%iret);

    iret = NormalOfMatrix(arr,irow,icol);
    print("Normal of matrix is : %.4f"%iret);

if __name__ == "__main__" :
    main();

File: TraceAndNormalMatrix/Find.cpp
#include<iostream>
#include<math.h>
using namespace std;

class Matrix
{
    private : 
        int **arr;
        int irow;
        int icol;

    public :
        Matrix(int **Arr,int iRow,int iCol) 
        {
            this->arr = Arr;
            this->irow = iRow;
            this->icol = iCol;
        }

    int TraceOfMatrix() 
    {
        int i = 0;
        int j = 0;
        int isum = 0;
        
        for(i = 0;i < this->irow;i++)
        {
            for(j = 0;j < this->icol;j++)
            {
                if(i == j)
                {
                    isum = isum + this->arr[i][j];
                }
            }
        }

        return isum;
    }

    int NormalOfMatrix() 
    {
        int i = 0;
        int j = 0;
        int isum = 0;
        int Sqrt = 0;

        for(i = 0;i < this->irow;i++)
        {
            for(j = 0;j < this->icol;j++)
            {
                isum = isum + this->arr[i][j];
            }
        }

        Sqrt = sqrt(isum);

        return Sqrt;
    }
};

int main() 
{
    int **arr = NULL;
    int irow = 0;
    int icol = 0;
    int iret = 0;
    int i = 0;
    int j = 0;

    cout<<"Jay Ganesh....\n";

    cout<<"Enter number of rows : ";
    cin>>irow;

    cout<<"Enter number of cols : ";
    cin>>icol;

    arr = (int**)malloc(irow * sizeof(int**));
    for(i = 0;i < irow;i++)
    {
        arr[i] = (int *)malloc(icol * sizeof(int));
    }
    
    cout<<"Enter elements of matrix \n";
    for(i = 0;i < irow;i++)
    {
        printf("-----Enter elements of %d row-----\n",(i + 1));
        for(j = 0;j < icol;j++)
        {
            cout<<"Enter element no "<<j + 1<<" : ";
            cin>>arr[i][j];
        }
    }   

    cout<<"Elements of matrix are : \n";
    for(i = 0;i < irow;i++)
    {
        for(j = 0;j < icol;j++)
        {
            cout<<arr[i][j]<<"\t";
        }
        cout<<"\n";
    }   

    Matrix *mobj = new Matrix(arr,irow,icol);

    iret = mobj->TraceOfMatrix();
    cout<<"Trace of matrix is : "<<iret<<"\n";

    iret = mobj->NormalOfMatrix();
    cout<<"Normal of matrix is approx : "<<iret<<"\n";

    sfor(i = 0;i < irow;i++)
    {
        free(arr[i]);
    }
    
    free(arr);

    return 0;
}